src/frontends/onnx/frontend/src/op/stft.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/stft.hpp"
+
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "onnx_import/core/null_node.hpp"
+#include "utils/common.hpp"
+#include "utils/dft.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_17 {
+
+OutputVector stft(const Node& node) {
+    const OutputVector ng_inputs{node.get_ng_inputs()};
+    auto signal = ng_inputs.at(0);
+    const auto dft_length_provided = ng_inputs.size() > 3 && !ngraph::op::is_null(ng_inputs[3]);
+    const auto onesided = node.get_attribute_value<int64_t>("onesided", 1);
+    const int64_t axis = 1;
+
+    const auto& frame_step_node = ng_inputs.at(1);
+    CHECK_VALID_NODE(node,
+                     ngraph::op::is_constant(frame_step_node.get_node_shared_ptr()) &&
+                         ov::shape_size(frame_step_node.get_shape()) <= 1,
+                     "frame_step input must be a scalar or Shape{1} constant.");
+    const auto frame_step =
+        ov::as_type_ptr<default_opset::Constant>(frame_step_node.get_node_shared_ptr())->cast_vector<int64_t>()[0];
+    const auto signal_param_shape = signal.get_partial_shape();
+    CHECK_VALID_NODE(node,
+                     signal_param_shape.rank().is_static() && signal_param_shape.rank().get_length() == 3 &&
+                         signal_param_shape.is_static(),
+                     "Shape of signal input must be static with the rank equal to 3.");
```

Comment:
If the whole shape is required to be static there is no need to check `signal_param_shape.rank().is_static()` and `signal_param_shape.is_static()` separately, it can be just `signal_param_shape.is_static()` because static shape must have static rank.
```suggestion
    CHECK_VALID_NODE(node,
                     signal_param_shape.is_static() && signal_param_shape.size() == 3,
                     "Shape of signal input must be static with the rank equal to 3.");
```

Reply:
that's right, thanks, done